modules/identity.go

```diff
@@ -0,0 +1,37 @@
+package modules
+
+//go:generate mkdir -p stubs
+//go:generate zbusc -module identity -version 0.0.1 -name manager -package stubs github.com/threefoldtech/zosv2/modules+IdentityManager stubs/identity_stub.go
+
+// Identifier is the interface that defines
+// how an object can be used an identity
+type Identifier interface {
+	Identity() string
+}
+
+// StrIdentifier is a helper type that implement the Identifier interface
+// on top of simple string
+type StrIdentifier string
+
+// Identity implements the Identifier interface
+func (s StrIdentifier) Identity() string {
+	return string(s)
+}
+
+// IdentityManager interface.
+type IdentityManager interface {
+	// NodeID returns the node id (public key)
+	NodeID() StrIdentifier
```

Comment:
```suggestion
	NodeID() Identifier
```
would be better, but does zbus supports interface as return value ?

Reply:
No, because it will not know what underlying type to use for the deserialization of the wire data. Hence u need to use a final type. Plus identity should never expose the private key, hence StrIdentifier is always the only possible type to return.

Reply:
ok got it.